sympy/functions/special/bessel.py

```diff
@@ -693,66 +693,67 @@ def _eval_nseries(self, x, n, logx, cdir=0):
+                #Reference: https://functions.wolfram.com/Bessel-TypeFunctions/BesselK/06/01/04/01/02/0008/ (only for integer order)
+                newn = ceiling(n/exp)
+                bn = besseli(nu, z)
+                a = ((-1)**(nu - 1)*log(z/2)*bn)._eval_nseries(x, n, logx, cdir)
+
+                b, c = [], []
+                o = Order(x**n, x)
+                r = (z/2)._eval_nseries(x, n, logx, cdir).removeO()
+                if r is S.Zero:
+                    return Order(log(z) + z**(-nu) + z**nu)
+                t = (_mexpand(r**2) + o).removeO()
+
+                if nu > S.Zero:
+                    term = r**(-nu)*factorial(nu - 1)/2
                     b.append(term)
-
-            p = r**nu*(-1)**nu/(2*factorial(nu))
-            term = p*(digamma(nu + 1) - S.EulerGamma)
-            c.append(term)
-            for k in range(1, (newn + 1)//2):
-                p *= t/(k*(k + nu))
-                p = (_mexpand(p) + o).removeO()
-                term = p*(digamma(k + nu + 1) + digamma(k + 1))
+                    for k in range(1, nu):
+                        denom = (k - nu)*k
+                        if denom == S.Zero:
+                            term *= t/k
+                        else:
+                            term *= t/denom
+                        term = (_mexpand(term) + o).removeO()
+                        b.append(term)
+
+                p = r**nu*(-1)**nu/(2*factorial(nu))
+                term = p*(digamma(nu + 1) - S.EulerGamma)
                 c.append(term)
-            return a + Add(*b) + Add(*c) # Order term comes from a
-        elif exp.is_positive:
-            #Reference: https://functions.wolfram.com/Bessel-TypeFunctions/BesselK/06/01/04/01/01/0003/ (only for non-integer order)
-            r = (z/2)._eval_nseries(x, n, logx, cdir).removeO()
-            if r is S.Zero:
-                return Order(z**(-nu), x)
-
-            newn_a = ceiling((n+nu)/exp)
-            newn_b = ceiling((n-nu)/exp)
-
-            a, b = [], []
-            for k in range((newn_a+1)//2):
-                a.append(
-                    gamma(nu)*r**(2*k-nu)/(2*RisingFactorial(1-nu, k)*factorial(k))
-                )
-            for k in range((newn_b+1)//2):
-                b.append(
-                    gamma(-nu)*r**(2*k+nu)/(2*RisingFactorial(nu+1, k)*factorial(k))
-                )
-            return Add(*a) + Add(*b) + Order(x**n, x)
+                for k in range(1, (newn + 1)//2):
+                    p *= t/(k*(k + nu))
+                    p = (_mexpand(p) + o).removeO()
+                    term = p*(digamma(k + nu + 1) + digamma(k + 1))
+                    c.append(term)
+                return a + Add(*b) + Add(*c) # Order term comes from a
+            else:
+                #Reference: https://functions.wolfram.com/Bessel-TypeFunctions/BesselK/06/01/04/01/01/0003/ (only for non-integer order)
+                r = (z/2)._eval_nseries(x, n, logx, cdir).removeO()
```

Comment:
I see the variable `r` being computed is similar for both if-else branches. This duplication can be taken care of.

Reply:
Done. I have also added a call to `_mexpand` on each term of the series (just like in the integer case).